atomic_reactor/plugins/post_export_operator_manifests.py

```diff
@@ -0,0 +1,139 @@
+from atomic_reactor.plugin import PostBuildPlugin
+from atomic_reactor.util import df_parser, is_scratch_build, get_platforms
+from docker.errors import APIError
+from osbs.utils import Labels
+from platform import machine
+
+MANIFESTS_DIR_NAME = 'manifests'
+OPERATORS_ARCHIVE_NAME, OPERATORS_ARCHIVE_EXT = OPERATOR_MANIFESTS_ARCHIVE.rsplit('.', 1)
+
+
+class ExportOperatorManifestsPlugin(PostBuildPlugin):
+    """
+    Export operator manifest files
+
+    Fetch and archive operator manifest files from image so they can be
+    uploaded to koji.
+    """
+
+    key = PLUGIN_EXPORT_OPERATOR_MANIFESTS_KEY
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow, operator_manifests_extract_platform=None, platform=None):
+        super(ExportOperatorManifestsPlugin, self).__init__(tasker, workflow)
+
+        self.operator_manifests_extract_platform = operator_manifests_extract_platform
+        if platform:
+            self.platform = platform
+        else:
+            self.platform = machine()
+
+    def has_operator_manifest(self):
+        """
+        Check if Dockerfile sets the operator manifest label
+
+        :return: bool
+        """
+        dockerfile = df_parser(self.workflow.builder.df_path, workflow=self.workflow)
+        labels = Labels(dockerfile.labels)
+        try:
+            _, operator_label = labels.get_name_and_value(Labels.LABEL_TYPE_OPERATOR_MANIFESTS)
+        except KeyError:
+            operator_label = 'false'
+        return operator_label.lower() == 'true'
+
+    def is_orchestrator(self):
+        """
+        Check if the plugin is running in orchestrator.
+
+        :return: bool
+        """
+        if get_platforms(self.workflow):
+            return True
+        return False
+
+    def should_run(self):
+        """
+        Check if the plugin should run or skip execution.
+
+        :return: bool, False if plugin should skip execution
+        """
+        if self.is_orchestrator():
+            self.log.warning("%s plugin set to run on orchestrator. Skipping", self.key)
+            return False
+        if self.operator_manifests_extract_platform != self.platform:
+            self.log.info("Only platform [%s] will upload operators metadata. Skipping",
+                          self.operator_manifests_extract_platform)
+            return False
+        if is_scratch_build():
+            self.log.info("Scratch build. Skipping")
+            return False
+        if not self.has_operator_manifest():
+            self.log.info("Operator manifests label not set in Dockerfile. Skipping")
+            return False
+        return True
+
+    def run(self):
+        """
+        Run the plugin.
+        """
+        if not self.should_run():
+            return
+        temp_dir = tempfile.mkdtemp()
+        manifests_zipfile_path = self.extract_manifests(temp_dir)
+        return manifests_zipfile_path
+
+    def extract_manifests(self, workdir):
```

Comment:
is there any reason this is a separate function instead of part of run?  It seems unnecessary to call a function here instead of just performing the same sequence of instructions.

Reply:
I wanted to avoid having a bloated run method, but in the way it was written, it woudn't make much sense to extract the method, as you pointed. I merged it into the `run` method as suggested.